### PR TITLE
Port/1.10 sign gui

### PR DESCRIPTION
**Pull request by Raycoms**
# Ports proposed in this pull request:
- BlockParticleMessage
- Citizen Info sign
- New TownHall GUI

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/blockout/controls/TextField.java [hunk 1]
```diff
@@ -315,10 +317,10 @@ public void onFocus()
         cursorBlinkCounter = 0;
     }
 
-    @Override
     /**
      * Draw itself at positions mx and my.
      */
+    @Override
     protected void drawSelf(final int mx, final int my)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/blockout/controls/TextField.java]: MAJOR The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
MAJOR Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

File: src/main/java/com/minecolonies/coremod/client/gui/WindowHomeBuilding.java [hunk 2]
```diff
@@ -1,24 +1,146 @@
...
+
+        refreshView();
+    }
+
+    /**
+     * Action when an assign button is clicked.
+     *
+     * @param button the clicked button.
+     */
+    private void assignClicked(@NotNull final Button button)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/client/gui/WindowHomeBuilding.java]: MAJOR Remove this unused method parameter "button". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1172)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 3]
```diff
@@ -0,0 +1,278 @@
...
+    
+    /**
+     * Convert the given metadata into a BlockState for this Block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState getStateFromMeta(final int meta)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+        return this.getDefaultState().withProperty(FACING, enumfacing);
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SideOnly(Side.CLIENT)
+    @Override
+    @Deprecated
+    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+        setLightOpacity(LIGHT_OPACITY);
+        blockMaterial.setReplaceable();
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getCollisionBoundingBox(IBlockState blockState, World worldIn, BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTape.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 9]
```diff
@@ -441,44 +680,113 @@ public int getID()
...
-     * Increment the mobs killed by this colony.
-     * <p>
-     * Will award achievements for mobs killed.
+     * Get the amount of statistic.
+     *
+     * @param statistic the statistic.
+     * @return amount of statistic.
      */
-    public void incrementMobsKilled()
+    private int getStatisticAmount(@NotNull String statistic)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MAJOR Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 10]
```diff
@@ -565,43 +878,61 @@ public void onServerTick(@NotNull final TickEvent.ServerTickEvent event)
     /**
      * Update Subscribers with Colony, Citizen, and AbstractBuilding Views.
      */
-    public void updateSubscribers()
+    private void updateSubscribers()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: MAJOR Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

File: src/main/java/com/minecolonies/coremod/colony/Colony.java [hunk 11]
```diff
@@ -643,6 +981,7 @@ public void updateSubscribers()
         isBuildingsDirty = false;
         permissions.clearDirty();
 
+        // TODO: look into potentially keeping lists as well
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/Colony.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 12]
```diff
@@ -0,0 +1,339 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 13]
```diff
@@ -0,0 +1,339 @@
...
+        return null;
+    }
+    
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+
+    @Override
+    @Deprecated
+    public AxisAlignedBB getCollisionBoundingBox(IBlockState blockState, World worldIn, BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 14]
```diff
@@ -0,0 +1,339 @@
...
+
+    /**
+     * Convert the given metadata into a BlockState for this Block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState getStateFromMeta(final int meta)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 15]
```diff
@@ -0,0 +1,339 @@
...
+        return this.getDefaultState().withProperty(FACING, enumfacing);
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SideOnly(Side.CLIENT)
+    @Override
+    @Deprecated
+    public boolean shouldSideBeRendered(IBlockState blockState, IBlockAccess blockAccess, BlockPos pos, EnumFacing side)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 16]
```diff
@@ -0,0 +1,339 @@
...
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java [hunk 17]
```diff
@@ -0,0 +1,339 @@
...
+    {
+        return true;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public AxisAlignedBB getBoundingBox(final IBlockState state, final IBlockAccess source, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockConstructionTapeCorner.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/colony/ColonyList.java [hunk 18]
```diff
@@ -0,0 +1,272 @@
...
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+
+//TODO extend list/collection
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyList.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/colony/ColonyList.java [hunk 19]
```diff
@@ -0,0 +1,272 @@
...
+ * Data structure for storing colonies, optimized for performance.
+ *
+ * @author Colton
+ *
+ * @param <T> Type of IColony (Colony or ColonyView)
+ */
+public final class ColonyList<T extends IColony> implements Iterable<T>
+{
+    @VisibleForTesting
+    static final int INITIAL_SIZE = 16;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyList.java]: MINOR Explicitly declare the visibility for "INITIAL_SIZE". [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2039)

File: src/main/java/com/minecolonies/coremod/colony/ColonyList.java [hunk 20]
```diff
@@ -0,0 +1,272 @@
...
+    {
+        if (list[id] == null)
+        {
+            Log.getLogger().warn("Tried to remove colony with id=%d, but it didn't exist.", id);
+        }
+
+        size--;
+        list[id] = null;
+
+        if (!nullIndices.contains(id))
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/ColonyList.java]: MINOR This call to "contains()" may be a performance hot spot if the collection is large. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

File: src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java [hunk 21]
```diff
@@ -57,10 +59,10 @@ public BlockBarrel()
     }
 
     //todo: register block with new method
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockBarrel.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 22]
```diff
@@ -0,0 +1,98 @@
...
+        return false;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @SuppressWarnings(DEPRECATION)
+    @Override
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 23]
```diff
@@ -0,0 +1,98 @@
...
+    }
+
+    /**
+     * Used to determine ambient occlusion and culling when rebuilding chunks
+     * for render.
+     * @deprecated
+     *
+     * @return true
+     */
+    //todo: remove once we no longer need to support this
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java [hunk 24]
```diff
@@ -0,0 +1,98 @@
...
+    @SuppressWarnings(DEPRECATION)
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 25]
```diff
@@ -0,0 +1,449 @@
...
+    {
+        final EnumFacing enumFacing = EnumFacing.getHorizontal(meta);
+        return this.getDefaultState().withProperty(FACING, enumFacing);
+    }
+
+    /**
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Override
+    public IBlockState getActualState(final IBlockState state, final IBlockAccess worldIn, final BlockPos pos)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: MAJOR Reduce the number of returns of this method 7, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
MAJOR Add the missing @Deprecated annotation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 26]
```diff
@@ -0,0 +1,449 @@
...
+        return state.withProperty(FACING, rot.rotate(state.getValue(FACING)));
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withMirror(@NotNull final IBlockState state, final Mirror mirrorIn)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 27]
```diff
@@ -0,0 +1,449 @@
...
+
+    @NotNull
+    @Override
+    @SideOnly(Side.CLIENT)
+    public BlockRenderLayer getBlockLayer()
+    {
+        return BlockRenderLayer.SOLID;
+    }
+
+    public enum EnumType implements IStringSerializable
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: MAJOR Reduce this class from 51 to the maximum allowed 30 or externalize it in a public class. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 28]
```diff
@@ -0,0 +1,449 @@
...
+    {
+        return new TileEntityRack();
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 29]
```diff
@@ -0,0 +1,449 @@
...
+    {
+        return false;
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 30]
```diff
@@ -0,0 +1,449 @@
...
+
+    /**
+     * Convert the BlockState into the correct metadata value.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 31]
```diff
@@ -0,0 +1,449 @@
...
+
+        worldIn.setBlockState(pos, tempState, 2);
+    }
+
+    /**
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Deprecated
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 32]
```diff
@@ -0,0 +1,449 @@
...
+        return state.getValue(FACING).getHorizontalIndex();
+    }
+
+    /**
+     * On the change of a neighbor block.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    public void neighborChanged(final IBlockState state, final World worldIn, final BlockPos pos, final Block blockIn)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: MAJOR Add the missing @Deprecated annotation. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java [hunk 33]
```diff
@@ -0,0 +1,449 @@
...
+    {
+        return state.withRotation(mirrorIn.toRotation(state.getValue(FACING)));
+    }
+
+    /**
+     * @deprecated (Remove this as soon as minecraft offers anything better).
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

File: src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java [hunk 34]
```diff
@@ -12,21 +12,28 @@
...
+    public static final Block blockHutWareHouse           = new BlockHutWareHouse();
+    public static final Block blockConstructionTape       = new BlockConstructionTape();
+    public static final Block blockConstructionTapeCorner = new BlockConstructionTapeCorner();
+    public static final Block blockTimberFrame            = new BlockTimberFrame();
+    public static final Block blockRack                   = new BlockMinecoloniesRack();
+    public static final Block blockWayPoint               = new BlockWaypoint();
+
 
     // Deactivated for now
     // public static final Block blockBarrel        = new BlockBarrel();
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/blocks/ModBlocks.java]: MINOR This block of commented-out lines of code should be removed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/main/java/com/minecolonies/coremod/client/gui/GuiFactory.java [hunk 35]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        return Collections.emptySet();
+    }
+    
+    /**
+     * @deprecated This was never fully implemented and will be removed in the future.
+     */
+    @Override
+    @Deprecated
+    public RuntimeOptionGuiHandler getHandlerFor(RuntimeOptionCategoryElement element)
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/client/gui/GuiFactory.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 316 issues
* ![MAJOR][MAJOR] 153 major
* ![MINOR][MINOR] 95 minor
* ![INFO][INFO] 68 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Pane.java#L19](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/Pane.java#L19): class "Pane" has 47 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [PaneParams.java#L349](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/PaneParams.java#L349): Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L349](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/PaneParams.java#L349): Assign this magic number 100 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L420](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/PaneParams.java#L420): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L439](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/PaneParams.java#L439): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L439](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/PaneParams.java#L439): Assign this magic number 8 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L444](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/PaneParams.java#L444): Assign this magic number 255 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [PaneParams.java#L444](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/PaneParams.java#L444): Assign this magic number 24 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L563](https://github.com/Minecolonies/minecolonies/blob/b59e7f660b5076f365692e2a23385375809cb4d2/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L563): Extract this nested ternary operation into an independent statement. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
@OrionDevelopment @Asherslab please ignore the codacity errors, this is just a port.

**Comment by Asherslab** [comment]
:shipit:



[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1409/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Review by marchermans — APPROVED** [review]

**Pull request merged**